### PR TITLE
Doc updates

### LINKED ISSUE #1216: [ISSUE] Can't manage sql permissions on workspace's hive_metastore
### Configuration
```hcl
resource "databricks_sql_permissions" "permissions" {
  catalog = true

  privilege_assignments {
    principal  = "hive_metastore_users"
    privileges = ["USAGE", "CREATE", "SELECT", "MODIFY"]
  }
}
```

### Expected Behavior
Grant "USAGE", "CREATE", "SELECT", "MODIFY" privileges to principal "hive_metastore_users" on workspace's local hive_metastore

### Steps to Reproduce
Please list the steps required to reproduce the issue, for example:
1. `terraform apply`

### Terraform and provider versions
databricks provider 0.5.4

### Notes
Unity Catalog is activated on this workspace (although what i'm dealing with here is indeed the workspace's own hive metastore)

### Cluster logs
```
...
22/03/23 15:16:04 INFO DriverDaemon: Starting driver daemon...
22/03/23 15:16:04 INFO SparkConfUtils$: Customize spark config according to file /tmp/custom-spark.conf
22/03/23 15:16:04 WARN SparkConf: The configuration key 'spark.akka.frameSize' has been deprecated as of Spark 1.6 and may be removed in the future. Please use the new key 'spark.rpc.message.maxSize' instead.
22/03/23 15:16:04 INFO DriverDaemon$: Attempting to run: 'enable iptables restrictions for Python'
22/03/23 15:16:05 INFO DriverDaemon$: Attempting to run: 'set up filesystem permissions for Python'
22/03/23 15:16:13 INFO DriverDaemon$: Not configuring RStudio Daemon due to process isolation being enabled
22/03/23 15:16:13 INFO DriverDaemon$$anon$1: Message out thread ready
22/03/23 15:16:13 INFO Server: jetty-9.4.43.v20210629; built: 2021-06-30T11:07:22.254Z; git: 526006ecfa3af7f1a27ef3a288e2bef7ea9dd7e8; jvm 1.8.0_302-b08
22/03/23 15:16:13 INFO AbstractConnector: Started ServerConnector@4ee80a94{HTTP/1.1, (http/1.1)}{0.0.0.0:6061}
22/03/23 15:16:13 INFO Server: Started @39775ms
22/03/23 15:16:13 INFO DriverDaemon: Driver daemon started.
22/03/23 15:16:14 WARN SQLConf: The SQL config 'spark.sql.hive.convertCTAS' has been deprecated in Spark v3.1 and may be removed in the future. Set 'spark.sql.legacy.createHiveTableByDefault' to false instead.
22/03/23 15:16:14 WARN SQLConf: The SQL config 'spark.sql.hive.convertCTAS' has been deprecated in Spark v3.1 and may be removed in the future. Set 'spark.sql.legacy.createHiveTableByDefault' to false instead.
22/03/23 15:16:15 INFO DriverCorral: Loading the root classloader
22/03/23 15:16:15 INFO DriverCorral: Starting sql repl ReplId-5a32f-09fc8-c75ef
22/03/23 15:16:15 INFO DriverCorral: Starting sql repl ReplId-51c66-bc70c-e5f4e-0
22/03/23 15:16:15 INFO DriverCorral: Starting sql repl ReplId-709fc-9921d-57f98-7
22/03/23 15:16:15 INFO DriverCorral: Starting sql repl ReplId-2572e-6c0cf-40855-5
22/03/23 15:16:15 INFO DriverCorral: Starting sql repl ReplId-78129-45ef2-6ec20-2
22/03/23 15:16:15 INFO SQLDriverWrapper: setupRepl:ReplId-5a32f-09fc8-c75ef: finished to load
22/03/23 15:16:15 INFO SQLDriverWrapper: setupRepl:ReplId-2572e-6c0cf-40855-5: finished to load
22/03/23 15:16:15 INFO SQLDriverWrapper: setupRepl:ReplId-709fc-9921d-57f98-7: finished to load
22/03/23 15:16:15 INFO SQLDriverWrapper: setupRepl:ReplId-78129-45ef2-6ec20-2: finished to load
22/03/23 15:16:15 INFO SQLDriverWrapper: setupRepl:ReplId-51c66-bc70c-e5f4e-0: finished to load
22/03/23 15:16:38 WARN SQLConf: The SQL config 'spark.sql.hive.convertCTAS' has been deprecated in Spark v3.1 and may be removed in the future. Set 'spark.sql.legacy.createHiveTableByDefault' to false instead.
22/03/23 15:16:38 WARN SQLConf: The SQL config 'spark.sql.hive.convertCTAS' has been deprecated in Spark v3.1 and may be removed in the future. Set 'spark.sql.legacy.createHiveTableByDefault' to false instead.
22/03/23 15:16:38 INFO DriverCorral: Starting sql repl ReplId-7e3e3-a44e4-e3042-6
22/03/23 15:16:38 INFO SQLDriverWrapper: setupRepl:ReplId-7e3e3-a44e4-e3042-6: finished to load
22/03/23 15:16:39 INFO ProgressReporter$: Added result fetcher for 9096772364907250726_7707065032651177604_64d0ce73-5660-48a3-ab28-48282cc0f069
22/03/23 15:16:40 ERROR SQLDriverLocal: Error in SQL query: SHOW GRANT ON CATALOG 

org.apache.spark.sql.AnalysisException: For unity catalog, please specify the catalog name explicitly. E.g. SHOW GRANT `your.address@email.com` ON CATALOG main
	at com.databricks.sql.managedcatalog.ManagedCatalogErrors$.shouldSpecifyTheCatalogName(ManagedCatalogErrors.scala:75)
	at com.databricks.sql.acl.CatalogSecurableIdentifier.toV1orV2Securable(statements.scala:57)
	at com.databricks.sql.acl.ResolvePermissionManagement.apply(ResolvePermissionManagement.scala:53)
	at com.databricks.sql.acl.ResolvePermissionManagement.apply(ResolvePermissionManagement.scala:34)
	at org.apache.spark.sql.catalyst.rules.RuleExecutor.$anonfun$execute$3(RuleExecutor.scala:216)
	at com.databricks.spark.util.FrameProfiler$.record(FrameProfiler.scala:80)
	...
```

### PR DESCRIPTION
**Pull request by nkvuong**
- Clarify edge case errors when creating `databricks_sql_permissions` on UC workspaces
- Add encryption and block public acl on root S3 bucket in E2 guides
Closes #1216

### CODE REVIEW

File: docs/guides/unity-catalog.md [hunk 1]
```diff
@@ -145,12 +145,21 @@ resource "aws_s3_bucket" "metastore" {
...
 }
 
-resource "aws_s3_bucket_public_access_block" "root_storage_bucket" {
+resource "aws_s3_bucket_public_access_block" "metastore" {
   bucket             = aws_s3_bucket.metastore.id
   ignore_public_acls = true
   depends_on         = [aws_s3_bucket.metastore]
 }
 
+resource "aws_s3_bucket_public_access_block" "metastore" {
```
Comment (nfx) [inline docs/guides/unity-catalog.md:154]: here we've got two `aws_s3_bucket_public_access_block.metastore` resources. make it one again :)

File: docs/resources/mws_workspaces.md [hunk 2]
```diff
@@ -141,9 +141,23 @@ resource "aws_s3_bucket" "root_storage_bucket" {
...
+
 resource "aws_s3_bucket_public_access_block" "root_storage_bucket" {
-  bucket             = aws_s3_bucket.root_storage_bucket.id
-  ignore_public_acls = true
+  bucket                  = aws_s3_bucket.root_bucket.id
+  block_public_acls       = true
+  block_public_policy     = true
+  ignore_public_acls      = true
+  restrict_public_buckets = true
+  depends_on              = [aws_s3_bucket.root_bucket]
```
Comment (nfx) [inline docs/resources/mws_workspaces.md:160]:
```suggestion
  depends_on              = [aws_s3_bucket.root_storage_bucket]
```

File: docs/resources/sql_permissions.md [hunk 3]
```diff
@@ -5,6 +5,8 @@ subcategory: "Security"
 
 -> **Note** Please switch to [databricks_grants](grants.md) with Unity Catalog to manage data access, which provides better and faster way for managing data security. `databricks_grants` resource *doesn't require a technical cluster to perform operations*. `databricks_sql_permissions` will be removed, once Unity Catalog is Generally Available.
 
+-> **Note** On workspaces with Unity Catalog enabled, you may run into errors such as `Error: cannot create sql permissions: cannot read current grants: For unity catalog, please specify the catalog name explicitly. E.g. SHOW GRANT ``your.address@email.com`` ON CATALOG main`. This happens if your default workspace catalog was set to a UC catalog instead of `hive_metastore`. The workaround is to re-assign the metastore again with the default catalog set to be `hive_metastore`.
```
Comment (nfx) [inline docs/resources/sql_permissions.md:8]: @nkvuong can you also provide that configuration of a single-node cluster with legacy table ACL enabled?

**Review by nfx — CHANGES_REQUESTED** [review]

File: docs/resources/sql_permissions.md (hunk 3 above)
Comment (nkvuong) [reply docs/resources/sql_permissions.md:8]: the issue is not actually with the cluster config (we still accept the legacy spark conf), it is to do with their default catalog not set to `hive_metastore`

Comment (nfx) [reply docs/resources/sql_permissions.md:8]: oh, okay.

**Review by nfx — APPROVED** [review]

**Pull request merged**